### PR TITLE
✨ Added ability to delete integrations

### PR DESCRIPTION
**Pull request by allouis**
Just a WIP to get some feedback on my :fire: ember stuff

Adds delete integration modal, and delete button on the integration page to trigger it.

Missing tests

### CODE REVIEW

File: app/controllers/settings/integration.js [hunk 1]
```diff
@@ -44,6 +44,11 @@ export default Controller.extend({
                     });
                 }
 
+                // if the model is deleted, allow the transition to complete
+                if (this.integration.get('isDeleted')) {
+                    return setTimeout(() => transition.retry(), 0);
```
Comment (allouis) [inline app/controllers/settings/integration.js]: Pretty sure this should be like `Ember.run` or something instead?

File: app/templates/settings/integration.hbs [hunk 2]
```diff
@@ -155,6 +155,9 @@
             </tfoot>
         </table>
     </div>
+    <button class="gh-btn gh-btn-red gh-btn-icon" {{action "confirmIntegrationDeletion" integration}}>
```
Comment (allouis) [inline app/templates/settings/integration.hbs]: This needs padding

File: app/controllers/settings/integration.js [hunk 3]
```diff
@@ -61,7 +66,20 @@ export default Controller.extend({
...
 
             return transition.retry();
+        },
+
+        deleteIntegration() {
+            this.integration.destroyRecord();
+        },
+
+        confirmIntegrationDeletion(integration) {
+            this.set('integrationToDelete', integration);
```
Comment (allouis) [inline app/controllers/settings/integration.js]: Similar pattern to webhooks, feels unnecessary as there's only ever one integration on a page. But it's useful to show the modal, maybe have a `showIntegrationDeleteModal` instead and set to true/false...?

File: app/templates/settings/integration.hbs (hunk 2 above)
Comment (allouis) [reply app/templates/settings/integration.hbs]: Or a `<span>` rather

File: app/controllers/settings/integration.js (hunk 3 above)
Comment (kevinansfield) [reply app/controllers/settings/integration.js]: Yeah, a true/false toggle would be clearer here I think

File: app/controllers/settings/integration.js (hunk 1 above)
Comment (kevinansfield) [reply app/controllers/settings/integration.js]: I don't think you need anything here, I would opt for putting a `!isDeleted` conditional in the [`willTransition` hook](https://github.com/TryGhost/Ghost-Admin/blob/190a710c9c9827bba1ba3d6711cbec5e793fcc37/app/routes/settings/integration.js#L49) that decides whether to show the unsaved changes modal

Comment (allouis) [reply app/controllers/settings/integration.js]: Ahhhhh okay :+1: i didn't even know about that- thank you thank you!

File: app/components/modal-delete-integration.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import ModalComponent from 'ghost-admin/components/modal-base';
+import {alias} from '@ember/object/computed';
+import {inject as service} from '@ember/service';
+import {task} from 'ember-concurrency';
+
+export default ModalComponent.extend({
+    router: service(),
+    notifications: service(),
+    integration: alias('model'),
+    actions: {
```
Comment (kevinansfield) [inline app/components/modal-delete-integration.js]: Stylistic thing but we generally keep everything spaced apart in Ghost-Admin. Single-line properties can be grouped such as the service imports, and one-line CPs but everything else has line breaks.

**Pull request merged**